3rdparty/CMakeLists.txt

```diff
@@ -459,6 +459,11 @@ install(
   DIRECTORY ${GLOG_INSTALL_DIR}/lib/
   DESTINATION ${MESOS_INSTALL_LIBRARIES})
 
+if (ENABLE_INSTALL_MODULE_DEPENDENCIES)
+  install(
+    DIRECTORY ${GLOG_INSTALL_DIR}/include
+    DESTINATION ${MESOS_INSTALL_LIBRARIES}/mesos/3rdparty)
+endif ()
```

Comment:
The autotools build system installs the 3rdparty dependencies by either running `make install` within the dependency's build directory, or by manually copying headers into the installation directory.

In the autotools build system, glog is installed with `make install`, as are picojson, protobuf, and zookeeper. Could you explain how you chose the install methods you're using below? I'm not familiar with the motivations for the precise methods used in our autotools `INSTALL_MODULE_DEPENDENCIES` implementation, so I'm just curious to figure out the best install method here. IIUC, the equivalent of the `make install` method here would be using `install(TARGETS ...)` to specify install directories for the 3rdparty builds?

Reply:
The motivation is to use a simplest possible method. In case of cmake that is `install(...)`, not really sure what is your concern here.

Reply:
Do we have a good understanding whether `install(TARGETS ...)` can be configured up to install headers:
 - of 3rdparties which have `INSTALL_COMMAND  ${CMAKE_NOOP}` in `ExternalProject_Add()`?
 - of 3rdparties that have **their** install step executed during build time?

If the answer is "no" in both cases, then `install(FILES ...`/`install(DIRECTORY ...` seems  to be the only option for installing headers, unfortunately.

Reply:
`install(TARGETS ...)` can be [configured](https://cmake.org/cmake/help/v3.13/command/install.html#installing-targets) with headers, if target defines [PUBLIC_HEADER](https://cmake.org/cmake/help/v3.13/prop_tgt/PUBLIC_HEADER.html#prop_tgt:PUBLIC_HEADER) or [PRIVATE_HEADER](https://cmake.org/cmake/help/v3.13/prop_tgt/PRIVATE_HEADER.html#prop_tgt:PRIVATE_HEADER). anyway you need to enumerate either with `PUBLIC_HEADER`, `PRIVATE_HEADER`, in `install(DIRECTORY ...)` or in `install(FILES ...)`.

Reply:
I don't think defining `PUBLIC_HEADER`/`PRIVATE_HEADER` for mesos/stout/libprocess at this point makes sense. This will require either specifying them manually, or enumerating via something like `files(GLOB ...)`; I don't see how this can add any value now. This can be done in future if needed.

And there seems to be no way to make `ExternalProject` define these, right? If yes, then `install(TARGETS ... )` doesn't help us with installing headers and we will be better off by just sticking with `install(FILES ...)`/`install(DIRECTORY ...)` for headers.